src/chunk.c

```diff
@@ -3376,47 +3367,107 @@ list_return_srf(FunctionCallInfo fcinfo)
 }
 
 static void
-drop_remote_chunks(FunctionCallInfo fcinfo, const Name schema_name, const Name table_name,
-				   List *data_node_oids)
+lock_referenced_tables(Oid table_relid)
 {
-	/* Wildcard drops not supported on distributed hypertables */
-	if (schema_name == NULL && table_name == NULL)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("cannot use wildcard to drop chunks on distributed hypertables"),
-				 errhint("Drop chunks on each distributed hypertable individually.")));
-
-	/* The schema name must be present when dropping remote chunks because the
-	 * search path on the connection is always set to pg_catalog. Thus, the
-	 * data node will not be able to resolve the same hypertables without the
-	 * schema. */
-	if (schema_name == NULL)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("schema required when dropping chunks on distributed hypertables")));
+	List *fk_relids = NIL;
+	ListCell *lf;
+	List *cachedfkeys = NIL;
+	Relation table_rel = table_open(table_relid, AccessShareLock);
 
-	/* Make sure the schema is set in case it was not given on the command line */
-	FC_ARG(fcinfo, 2) = NameGetDatum(schema_name);
-	FC_NULL(fcinfo, 2) = false;
+	/* this list is from the relcache and can disappear with a cache flush, so
+	 * no further catalog access till we save the fk relids */
+	cachedfkeys = RelationGetFKeyList(table_rel);
+	foreach (lf, cachedfkeys)
+	{
+		ForeignKeyCacheInfo *cachedfk = (ForeignKeyCacheInfo *) lfirst(lf);
+
+		/* conrelid should always be that of the table we're considering */
+		Assert(cachedfk->conrelid == RelationGetRelid(table_rel));
+		fk_relids = lappend_oid(fk_relids, cachedfk->confrelid);
+	}
+	table_close(table_rel, AccessShareLock);
 
-	ts_cm_functions->func_call_on_data_nodes(fcinfo, data_node_oids);
+	/* We have a FK between hypertable H and PAR. Hypertable H has number of
```

Comment:
I have been asked to move these comments around several times already, I'm not moving them again. It is trivial to go to the function and read the comment there.

Reply:
This comment is not meaningful at this place, since it describes semantics of the caller.

Since you complain that you were asked to move the comment several times, this indicates that there is some issue with the code and comment, but nobody can grasp it. Remember that you were providing similar hopeless reviews and now you on other side where you need patience, so the code results to be understandable and maintainable,.

Reply:
Comment moved.